### PR TITLE
Use only id and version to check if a read record is not changed in the validation in EXTRA_READ

### PR DESCRIPTION
**Pull request by brfrn169**
I don't think we need to compare all the values of a read record and a re-read record in the validation in EXTRA_READ. I think it's enough to compare `id` and `version` values. This PR makes it. Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java [hunk 1]
```diff
@@ -347,6 +348,18 @@ void toSerializableWithExtraRead(DistributedStorage storage)
...
+    if (latestResult.isPresent() != result.isPresent()) {
+      return true;
+    }
+    if (!latestResult.isPresent()) {
+      return false;
+    }
+    return !latestResult.get().getId().equals(result.get().getId())
+        || latestResult.get().getVersion() != result.get().getVersion();
+  }
+
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java:362]: This is the new comparison logic. It compares only `id` and `version` values of the read and re-read records.

Comment (feeblefakie) [reply core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java:362]: It's a super rare case, but a read record could be deleted, created, and updated several times to get to the read version by other transactions, so we need to compare the transaction ID as well?

**Review by feeblefakie — COMMENTED** [review]
Thank you!
I left one question. PTAL!

File: core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java (hunk 1 above)
Comment (brfrn169) [reply core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java:362]: Thank you for taking a look at it!

Yes, I think the logic already compare the transaction ID (`!latestResult.get().getId().equals(result.get().getId())`). I think the logic compares both the transaction ID and version. That's not enough?
```java
    return !latestResult.get().getId().equals(result.get().getId())
        || latestResult.get().getVersion() != result.get().getVersion();
```

Comment (feeblefakie) [reply core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java:362]: Oh, sorry! I was a bit confused with Scalar DL. Thank you!

**Review by feeblefakie — DISMISSED** [review]
LGTM! Thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**